File: app/src/main/java/net/chris/demo/guardian/ui/recycler/SectionViewHolder.java
package net.chris.demo.guardian.ui.recycler;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import net.chris.demo.guardian.BR;

public class SectionViewHolder extends RecyclerView.ViewHolder {

    private final SectionListItem item;
    private final ViewDataBinding binding;

    public SectionViewHolder(View itemView, SectionListItem item) {
        super(itemView);
        binding = DataBindingUtil.bind(itemView);
        this.item = item;
    }

    public void bind(SectionListItem item) {
        binding.setVariable(BR.item, item);
        binding.executePendingBindings();
    }

    public SectionListItem getItem() {
        return item;
    }
}

File: app/src/main/kotlin/net/chris/demo/guardian/ui/recycler/SectionViewHolder.kt
package net.chris.demo.guardian.ui.recycler

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView
import android.view.View

import net.chris.demo.guardian.BR

class SectionViewHolder(itemView: View, val item: SectionListItem) : RecyclerView.ViewHolder(itemView) {
    private val binding: ViewDataBinding = DataBindingUtil.bind(itemView)

    fun bind(item: SectionListItem) {
        binding.setVariable(BR.item, item)
        binding.executePendingBindings()
    }
}
